libraries/resource_providers/api200/volume_template_provider.rb

```diff
@@ -13,28 +13,29 @@ module OneviewCookbook
   module API200
     # VolumeTemplate API200 provider
     class VolumeTemplateProvider < ResourceProvider
+      include OneviewCookbook::Validations::Presence
+
       # Loads the VolumeTemplate with all the external resources (if needed)
       def load_resource_with_associated_resources
-        raise "Unspecified property: 'storage_system'. Please set it before attempting this action." unless @new_resource.storage_system
-        raise "Unspecified property: 'storage_pool'. Please set it before attempting this action." unless @new_resource.storage_pool
+        validate_presence_of(:storage_system, :storage_pool)
+        storage_system = load_storage_system
```

Comment:
can be likely substituted by:
```ruby
storage_system_data = {credentials: { ip_hostname: @new_resource.storage_system }, name: @new_resource.storage_system}
storage_system = load_resource(:StorageSystem, storage_system_data)
```
And the method `load_storage_system` can then be dropped.